marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mastering/matching/matchable.sjs

```diff
@@ -15,7 +16,41 @@ class Matchable {
    * @since 5.8.0
    */
   baselineQuery() {
-    // TODO DHFPROD-8589
+    if (!this._baselineQuery) {
+      let firstBaseline;
+      if (this.matchStep.targetEntityType) {
+        const targetEntityType = this.matchStep.targetEntityType;
+        const tripleQuery = cts.tripleRangeQuery(null, sem.iri("http://www.w3.org/1999/02/22-rdf-syntax-ns#type"), sem.iri(targetEntityType))
+        // check for TDE for entity to be enabled
+        if (cts.exists(tripleQuery)) {
+          firstBaseline = tripleQuery;
+        } else {
+          firstBaseline = cts.collectionQuery(targetEntityType.substring(targetEntityType.lastIndexOf("/") + 1));
+        }
+      } else if (this.matchStep.collections && this.matchStep.collections.content) {
```

Comment:
Why are we doing this filter in the else?
My understanding is that if a collection or a query is included in the step we need add that filter to the target entity type filter.
Currently we have that filter (by collection or by query) and the interceptors as two separate items.
![image](https://user-images.githubusercontent.com/4117902/159560018-21dc91fd-9957-4adb-ade8-7d9aef1929b7.png)